pkg/apis/neptune/v1alpha1/federatedlearningjob_types.go

```diff
@@ -0,0 +1,152 @@
+}
+
+// TrrainingWorker describes the data a training worker should have
+type TrainingWorker struct {
+	NodeName   string             `json:"nodeName"`
+	WorkerSpec TrainingWorkerSpec `json:"workerSpec"`
+	Dataset    datasetRefer       `json:"dataset"`
+}
+
+// AggregationWorkerSpec is a description of a aggregationworker
+type AggregationWorkerSpec struct {
+	CommonWorkerSpec
+}
+
+// TrainingWorkerSpec is a description of a trainingworker
+type TrainingWorkerSpec struct {
+	CommonWorkerSpec
+}
+
+type datasetRefer struct {
+	Name string `json:"name"`
+}
+
+type modelRefer struct {
+	Name string `json:"name"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FederatedLearningJobList is a list of FederatedLearningJobs.
+type FederatedLearningJobList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []FederatedLearningJob `json:"items"`
+}
+
+// FLJobStatus represents the current state of a federatedlearning job.
+type FLJobStatus struct {
+
+	// The latest available observations of a federated job's current state.
+	// +optional
+	Conditions []FLJobCondition `json:"conditions,omitempty"`
+
+	// Represents time when the job was acknowledged by the job controller.
+	// It is not guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	// +optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// Represents time when the job was completed. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	// +optional
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// The number of actively running pods.
+	// +optional
+	Active int32 `json:"active"`
+
+	// The number of pods which reached phase Succeeded.
+	// +optional
+	Succeeded int32 `json:"succeeded"`
+
+	// The number of pods which reached phase Failed.
+	// +optional
+	Failed int32 `json:"failed"`
+
+	// The phase of the federatedlearning job.
+	// +optional
+	Phase FLJobPhase `json:"phase,omitempty"`
+}
+
+type FLJobConditionType string
+
+// These are valid conditions of a job.
+const (
+	// FLJobComplete means the job has completed its execution.
+	FLJobCondComplete FLJobConditionType = "Complete"
+	// FLJobFailed means the job has failed its execution.
+	FLJobCondFailed FLJobConditionType = "Failed"
+	// FLJobCondTraining means the job has been training.
+	FLJobCondTraining FLJobConditionType = "Training"
+)
+
+// FLJobCondition describes current state of a job.
+type FLJobCondition struct {
+	// Type of job condition, Complete or Failed.
+	Type FLJobConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status"`
+	// Last time the condition was checked.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
```

Comment:
rename to LastHeartbeatTime pls see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

Reply:
ok，i'll update this uniformly in next version

Reply:
remaind that crd yaml is also needed to be updated if you decide to update.